### PR TITLE
Feature/get pull request patch

### PR DESCRIPTION
**Pull request by TomMD**
The patch target, much like the diff target, is accessible just by changing the header.

### CODE REVIEW

**Comment by TomMD** [comment]
This PR, among others, keeps me stuck in a fork.  Assuming I update so that it applies cleanly is there anything else needed to get merged?

**Comment by phadej** [comment]
@TomMD thanks for the ping. Yes, I'll go through the PRs and make new release, let's see if this applies cleanly as is.

File: src/GitHub/Endpoints/PullRequests.hs [hunk 1]
```diff
@@ -60,6 +67,45 @@ pullRequestsForR user repo opts = pagedQuery
...
+-- | Obtain the diff of a pull request
+-- See <https://developer.github.com/v3/pulls/#get-a-single-pull-request>
+pullRequestDiff :: Name Owner -> Name Repo -> Id PullRequest -> IO (Either Error ByteString)
+pullRequestDiff = pullRequestDiff' Nothing
+
+-- | Query a single pull request to obtain the diff
+-- See <https://developer.github.com/v3/pulls/#get-a-single-pull-request>
+pullRequestDiffR :: Name Owner -> Name Repo -> Id PullRequest -> Request k ByteString
+pullRequestDiffR user repo prid =
+    RawHeaderQuery
```
Comment (phadej) [inline src/GitHub/Endpoints/PullRequests.hs]: is this header still required?

Comment (phadej) [reply src/GitHub/Endpoints/PullRequests.hs]: Ok. I see, we need this to get `ByteString`. I'd prefer a concrete sum type for `Diff`, `Patch` and sha from https://developer.github.com/v3/media/#commits-commit-comparison-and-pull-requests

**Comment by phadej** [comment]
Superseded by #350

**Pull request closed without merging**